### PR TITLE
Updating coupling profiles layout page

### PR DESCRIPTION
**Pull request by Jimmy-A-Caroli**
In **signprot/statistics** page:
- Updated Venn Diagram with grayscale for each selection
- Generated table to deploy intersection data derived from Venn Diagram
- Changed Cladogram Tree layout matching previously shown in the GPCRdb
- Added DrawCircles function to enable coupling profiles data into the Cladogram
- Minor tweaks in the References section
- Modified Menu system accordingly to fetch the new page
- Added data from Bouvier and Inoue in the data pool
- Added the option to select between "GTP+" (GTP or any 2 sources of data) or "Any 2 Sources" 

Minor tweaks:
- Adjusted jvenn.js library to match new Venn Diagram system
- Adjusted phylo_tree.js to match new DrawCircles function in Cladogram
- Minor bugfixes

### CODE REVIEW

File: common/phylogenetic_tree.py [hunk 1]
```diff
@@ -53,17 +53,27 @@ def update_exp_data(self, data):
...
     def get_nodes_dict(self, param):
-        return OrderedDict([
-            ('name', self.name),
-            ('value', self.get_value(param)),
-            ('color', self.color),
-            ('children', [
-                y.get_nodes_dict(param) for x,y in self.children.items() if self.children != OrderedDict()
-                ]),
-            ])
+        if param == None:
```
Comment (AJK-dev) [inline common/phylogenetic_tree.py:59]: Maybe add a comment explaining this switch and how it works

File: signprot/views.py [hunk 2]
```diff
@@ -423,6 +424,165 @@ def GProtein(request, dataset="GuideToPharma", render_part="both"):
...
+                    jsondata.pop(key, None)
+                else:
+                    jsondata[key] = ''.join(jsondata[key])
+
+                if len(jsondata_gtp_plus[key]) == 0:
+                    jsondata_gtp_plus.pop(key, None)
+                else:
+                    jsondata_gtp_plus[key] = ''.join(jsondata_gtp_plus[key])
+
+            context[slug_translate[slug]] = jsondata
```
Comment (AJK-dev) [inline signprot/views.py:548]: Beware of hardcoding any elements (e.g. Gs etc.)

File: static/home/js/gprotselectivitytree.js [hunk 3]
```diff
@@ -164,7 +164,7 @@ for (var x in selectivityinfo){
     var spacer = 8
 
 
-    if(selectivityinfo[x].indexOf("Gs family") >= 0){
+    if(selectivityinfo[x].indexOf("Gs") >= 0){
```
Comment (AJK-dev) [inline static/home/js/gprotselectivitytree.js:167]: I know this is a legacy function, but maybe merge the four if-statements into a single for loop.

**Review by AJK-dev — APPROVED** [review]
Few notes for future consideration. Everything looks good!
Great PR message 👍

File: signprot/views.py [hunk 4]
```diff
@@ -423,6 +424,165 @@ def GProtein(request, dataset="GuideToPharma", render_part="both"):
...
+            for gp in gproteins:
+                # Collect GTP
+                gtp_couplings = list(ProteinGProteinPair.objects.filter(protein__family__slug__startswith=slug, source="GuideToPharma", g_protein=gp)\
+                                .order_by("protein__entry_name")\
+                                .values_list("protein__entry_name", flat=True)\
+                                .distinct())
+
+                # Other coupling data with logmaxec50 greater than 0
+                # TODO - coupling source should not be hardcoded
+                other_couplings = list(ProteinGProteinPair.objects.filter(protein__family__slug__startswith=slug, source__in=["Aska2", "Bouvier2"])\
```
Comment (esguerra) [inline signprot/views.py:490]: Just a heads-up.
Aska2 and Bouvier2 don't exist any longer in the latest version.
They are now just Inoue and Bouvier.

Comment (AJK-dev) [reply signprot/views.py:490]: Thanks for the heads-up just fixed it with an exclude call instead of the other way round.

**Pull request merged**